### PR TITLE
Create a Quiz

### PR DESCRIPTION
**Pull request by Blokkinen**
PT Story: https://www.pivotaltracker.com/story/show/132559999

Changes proposed in this pull request:

* Create a Quiz Class
* Add attr code, name
* Display name and Code on the Quiz view
* Unit tests and cucumber

What I have learned working on this feature: 
 - Nifty new line of code for generating random number/letter strings.

Ready for review!

### CODE REVIEW

File: spec/models/quiz_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  describe 'Factory' do
+    it 'should have a valid Factory' do
+      expect(FactoryGirl.create(:quiz)).to be_valid
+    end
+  end
+
+  describe 'Code' do
+    let(:quiz) { FactoryGirl.create(:quiz) }
+    it 'should not have more than 4 characters' do
+      expect(quiz.code.length).to eq(4)
```
Comment (tochman) [inline spec/models/quiz_spec.rb]: We should add and test for that validation.
```
validates_length_of :code, is: 4
```

File: spec/factories/quizzes.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :quiz do
+    name "MyString"
+    code "MyString"
```
Comment (tochman) [inline spec/factories/quizzes.rb]: That should probably be 4 characters

File: features/step_definitions/quiz_steps.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Given(/^there is a quiz called "([^"]*)"$/) do |name|
+  FactoryGirl.create(:quiz, name: name)
+end
+
+Given(/^I am on the quiz page for "([^"]*)"$/) do |quiz_name|
+  @quiz = Quiz.find_by(name: quiz_name)
```
Comment (tochman) [inline features/step_definitions/quiz_steps.rb]: I don't mind this being set as an instance variable but I don't see a reason for it atm.

**Comment by tochman** [comment]
You guys can merge your own PR's if you like. Or have me and @diraulo do reviews and merges if you feel more comfortable with that.

**Comment by tochman** [comment]
Having a separate namespace for the quizmaster feels okay for me.

File: features/step_definitions/quiz_steps.rb (hunk 3 above)
Comment (AmberWilkie) [inline features/step_definitions/quiz_steps.rb]: Oh right, I was expecting `@quiz` to get sent to the view but it didn't... I don't actually know why.

**Comment by AmberWilkie** [comment]
I would prefer to have either @tochman or @diraulo look over code before it gets merged. You almost always have something useful to say.

File: spec/factories/quizzes.rb (hunk 2 above)
Comment (AmberWilkie) [inline spec/factories/quizzes.rb]: I just got rid of it. Our validation adds a code.

File: features/step_definitions/quiz_steps.rb (hunk 3 above)
Comment (AmberWilkie) [inline features/step_definitions/quiz_steps.rb]: @tochman any insights about that? Otherwise, I'm ready to take the WIP off this.

Comment (tochman) [inline features/step_definitions/quiz_steps.rb]: Well, it IS set by the controller and passed in to the view, although it has nothing to do with the tests. ;-)

**Comment by Blokkinen** [comment]
Or I could do it.. Review our code.. You trust me right? 😉

File: features/step_definitions/quiz_steps.rb (hunk 3 above)
Comment (tochman) [inline features/step_definitions/quiz_steps.rb]: All good. Just a misunderstanding.

**Pull request merged**